### PR TITLE
[FIX] New users aren't mapped to rooms with OAuth groups/channels map

### PR DESCRIPTION
**Pull request by carlosrodrigues94**
## Proposed changes (including videos or screenshots)

  This change deals with the problem with the new users coming from key cloak, this users were not being mapped to the correct channels on RC.

## Steps to test or reproduce
- Setup a KeyCloak VM and configure it on Rocketchat 
(https://docs.google.com/document/d/1cBohJ1XrLm6vFIdBkEo8sr7xI2huEBE-sCCxcDibZU0/edit)

- enable the setting to map users to channels and login on RC using keycloak, you should be redirected to a channel.

## Further comments

TC-69

### CODE REVIEW

File: apps/meteor/app/federation-v2/server/infrastructure/rocket-chat/hooks/index.ts
Comment (matheusbsilva137) [inline apps/meteor/app/federation-v2/server/infrastructure/rocket-chat/hooks/index.ts:40]:
```suggestion
				if (!params || !params.user || !params.inviter || !room) {
```

File: apps/meteor/app/federation-v2/server/infrastructure/rocket-chat/hooks/index.ts
Comment (matheusbsilva137) [inline apps/meteor/app/federation-v2/server/infrastructure/rocket-chat/hooks/index.ts:56]:
```suggestion
				if (!params || !params.user || !params.inviter || !room) {
```

**Review by matheusbsilva137 — CHANGES_REQUESTED** [review]

**Review by matheusbsilva137 — DISMISSED** [review]

File: apps/meteor/app/federation-v2/server/infrastructure/rocket-chat/hooks/index.ts
Comment (pierre-lehnen-rc) [inline apps/meteor/app/federation-v2/server/infrastructure/rocket-chat/hooks/index.ts:54]:
```suggestion
				if (!params || !params.user || !room) {
```
Inviter is optional.

File: apps/meteor/app/federation-v2/server/infrastructure/rocket-chat/hooks/index.ts
Comment (pierre-lehnen-rc) [inline apps/meteor/app/federation-v2/server/infrastructure/rocket-chat/hooks/index.ts:39]:
```suggestion
			(params: { user: IUser | string; inviter?: IUser }, room: IRoom): void => {
				if (!params || !params.user || !room) {
```
Inviter is optional.

**Review by pierre-lehnen-rc — CHANGES_REQUESTED** [review]
Change the `federation.beforeAddUserAToRoom` definition to list Inviter as optional, matching the definition of the addUserToRoom function.

File: apps/meteor/app/authorization/server/streamer/permissions/index.ts
Comment (pierre-lehnen-rc) [inline apps/meteor/app/authorization/server/streamer/permissions/index.ts:7]: updatedAt is supposed to already be a Date

File: apps/meteor/app/custom-oauth/server/custom_oauth_server.js
Comment (pierre-lehnen-rc) [inline apps/meteor/app/custom-oauth/server/custom_oauth_server.js:353]: Same issue as the other PR - once that one is merged it will conflict here and you can keep that PR's version

**Review by pierre-lehnen-rc — DISMISSED** [review]

File: apps/meteor/app/federation-v2/server/infrastructure/rocket-chat/hooks/index.ts
Comment (pierre-lehnen-rc) [inline apps/meteor/app/federation-v2/server/infrastructure/rocket-chat/hooks/index.ts:39]: Should the callback really not run when inviter is undefined?

**Comment by pierre-lehnen-rc** [comment]
Needs review from federation team as well.

File: apps/meteor/app/federation-v2/server/infrastructure/rocket-chat/hooks/index.ts
Comment (carlosrodrigues94) [reply apps/meteor/app/federation-v2/server/infrastructure/rocket-chat/hooks/index.ts:39]: yeah because for some reason this callback it was running when the user it is from keycloak

**Comment by carlosrodrigues94** [comment]
> Needs review from federation team as well.

this change it was already aligned with @MarcosSpessatto

File: apps/meteor/app/federation-v2/server/infrastructure/rocket-chat/hooks/index.ts
Comment (matheusbsilva137) [reply apps/meteor/app/federation-v2/server/infrastructure/rocket-chat/hooks/index.ts:39]: Hey @MarcosSpessatto is that ok to keep this change? I see it fixes the issue, but I guess this just always causes the callback not to run (otherwise we'd throw the `error-cant-add-federated-users` error since `isAddingANewExternalUser` would return true).

File: apps/meteor/app/authorization/server/streamer/permissions/index.ts
Comment (matheusbsilva137) [reply apps/meteor/app/authorization/server/streamer/permissions/index.ts]: Fixed!

File: apps/meteor/app/federation-v2/server/infrastructure/rocket-chat/hooks/index.ts
Comment (MarcosSpessatto) [reply apps/meteor/app/federation-v2/server/infrastructure/rocket-chat/hooks/index.ts:39]: We should remove the `!params.inviter` as @pierre-lehnen-rc said since we are not using the inviter inside the method handling this callback. If you remove the `!params.inviter`, is this throwing an error anyway?

Comment (matheusbsilva137) [reply apps/meteor/app/federation-v2/server/infrastructure/rocket-chat/hooks/index.ts]: Yeah, it was still throwing an error since we were sending the `internalUser` as a string. I changed some stuff so that an `IUser` is sent and this is now working without the `!params.inviter` check.

File: apps/meteor/app/federation-v2/server/infrastructure/rocket-chat/hooks/index.ts
Comment (MarcosSpessatto) [inline apps/meteor/app/federation-v2/server/infrastructure/rocket-chat/hooks/index.ts:54]: I believe we need to remove the `!params.inviter` from here.

**Review by MarcosSpessatto — CHANGES_REQUESTED** [review]

File: apps/meteor/app/federation-v2/server/infrastructure/rocket-chat/hooks/index.ts
Comment (matheusbsilva137) [reply apps/meteor/app/federation-v2/server/infrastructure/rocket-chat/hooks/index.ts]: Fixed!

**Review by MarcosSpessatto — APPROVED** [review]

**Pull request merged**
